src/broadcast/broadcast.rs

```diff
@@ -380,9 +380,12 @@ impl<N: NodeIdT> Broadcast<N> {
         let p = p.clone().unwrap();
         let echo_msg = Message::Echo(p.clone());
         let mut step = Step::default();
-        // TODO: iterator not correctly describing left of our_id. Use cycle method on
-        // iterator to correctly specify.
-        let left = self.netinfo.all_ids_except(self.proposer_id()).take(self.netinfo.num_correct()+self.fault_estimate);
+        // `N - 2f + g` node ids to the left of our_id (including our_id)
+        // after arranging all node ids in a circular list.
+        let left = self.netinfo.all_ids_except(self.proposer_id())
+                               .cycle()
+                               .skip_while(|x| *x == self.our_id())
+                               .take(self.netinfo.num_correct() - self.netinfo.num_faulty()+self.fault_estimate);
```

Comment:
`.skip_while(|x| *x == self.our_id())` would always skip 0 elements except if we're the first entry. You probably wanted to say `!=` and then skip one more?

But I think we can actually count the proposer towards the _N - 2f + g_, since it definitely does have the data.
The same goes for ourselves. So I'd do something like:
```rust
let left = self.netinfo.all_ids()
    .cycle()
    .skip_while(|x| *x != self.our_id())
    .take(self.netinfo.num_correct() - self.netinfo.num_faulty()+self.fault_estimate)
    .skip(1);
```
…and maybe in `send_echo` filter out the proposer?

Reply:
Yes it should be `!=` and then skip one for ourself. I didn't get why we would count the proposer towards the N - 2f + g though. Seems wasteful since he already has the data.

Sorry for all the unchecked code btw. I was changing laptops so committed everything I had without checking. Forgot that pull request would get updated. :sweat_smile: